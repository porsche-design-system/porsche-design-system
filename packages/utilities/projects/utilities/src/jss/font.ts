export type FontSize = 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge';
export type FontWeight = 'thin' | 'regular' | 'semibold' | 'bold';
export type FontSizeLineHeight = { fontSize: string; lineHeight: number };
type Font = {
  family: string;
  weight: { [key in FontWeight]: number };
  size: { [key in FontSize]: FontSizeLineHeight };
};

export const fontFamily = "'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif";
export const fontWeight: { [key in FontWeight]: number } = {
  thin: 100,
  regular: 400,
  semibold: 600,
  bold: 700,
};
export const fontSize: { [key in FontSize]: FontSizeLineHeight } = {
  xSmall: {
    fontSize: '0.75rem',
    lineHeight: 1.6666666667,
  },
  small: {
    fontSize: '1rem',
    lineHeight: 1.5,
  },
  medium: {
    fontSize: '1.5rem',
    lineHeight: 1.5,
  },
  large: {
    fontSize: '2.25rem',
    lineHeight: 1.3333333333,
  },
  xLarge: {
    fontSize: '3.25rem',
    lineHeight: 1.2307692308,
  },
};

export const font: Font = {
  family: fontFamily,
  weight: fontWeight,
  size: fontSize,
};
