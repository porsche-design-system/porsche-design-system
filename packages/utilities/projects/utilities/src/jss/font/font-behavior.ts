import type { FontBehavior } from './font-shared';

export const fontBehavior: FontBehavior = {
  overflowWrap: 'break-word',
  hyphens: 'auto',
  WebkitTextSizeAdjust: 'none', // stop iOS safari from adjusting font size when screen rotation is changing
  textSizeAdjust: 'none',
};
