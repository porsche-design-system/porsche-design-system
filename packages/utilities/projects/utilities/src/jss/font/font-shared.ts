export type FontSize = 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge';
export type FontWeight = 'thin' | 'regular' | 'semibold' | 'bold';
export type FontSizeLineHeight = { fontSize: string; lineHeight: number };
export type FontBehavior = {
  overflowWrap: string;
  hyphens: string;
  WebkitTextSizeAdjust: string;
  textSizeAdjust: string;
};
