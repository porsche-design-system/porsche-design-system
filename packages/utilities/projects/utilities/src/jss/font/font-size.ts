import type { FontSize, FontSizeLineHeight } from './font-shared';

export const fontSize: { [key in FontSize]: FontSizeLineHeight } = {
  xSmall: {
    fontSize: '0.75rem',
    lineHeight: 1.6666666667,
  },
  small: {
    fontSize: '1rem',
    lineHeight: 1.5,
  },
  medium: {
    fontSize: '1.5rem',
    lineHeight: 1.5,
  },
  large: {
    fontSize: '2.25rem',
    lineHeight: 1.3333333333,
  },
  xLarge: {
    fontSize: '3.25rem',
    lineHeight: 1.2307692308,
  },
};
