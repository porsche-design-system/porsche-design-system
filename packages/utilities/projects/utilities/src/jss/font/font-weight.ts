import type { FontWeight } from './shared';

export const fontWeight: { [key in FontWeight]: number } = {
  thin: 100,
  regular: 400,
  semibold: 600,
  bold: 700,
};
