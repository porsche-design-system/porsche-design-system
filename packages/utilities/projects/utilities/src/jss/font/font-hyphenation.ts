import { FontHyphenation } from './font-shared';

export const fontHyphenation: FontHyphenation = {
  hyphens: 'auto',
};
