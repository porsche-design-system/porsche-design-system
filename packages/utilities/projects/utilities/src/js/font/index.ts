import { fontBehavior } from './font-behavior';
import { fontFamily } from './font-family';
import { fontSize } from './font-size';
import { fontWeight } from './font-weight';
import { fontStyle } from './font-style';
import { fontVariant } from './font-variant';

export { fontBehavior, fontFamily, fontSize, fontWeight, fontStyle, fontVariant };
