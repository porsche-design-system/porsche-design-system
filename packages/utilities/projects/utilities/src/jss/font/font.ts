import type { FontSize, FontSizeLineHeight, FontWeight } from './font-shared';
import { fontFamily } from './font-family';
import { fontSize } from './font-size';
import { fontWeight } from './font-weight';

type Font = {
  family: string;
  weight: { [key in FontWeight]: number };
  size: { [key in FontSize]: FontSizeLineHeight };
};

const font: Font = {
  family: fontFamily,
  weight: fontWeight,
  size: fontSize,
};

export { font, fontFamily, fontSize, fontWeight, FontSize, FontSizeLineHeight, FontWeight };
