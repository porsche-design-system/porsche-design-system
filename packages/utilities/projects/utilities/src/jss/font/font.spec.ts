import * as fromFont from './font';

it('should provide all font exports', () => {
  expect(Object.keys(fromFont).length).toBe(7);
});

it.each<keyof typeof fromFont>(Object.keys(fromFont) as (keyof typeof fromFont)[])(
  'should contain correct values for %s',
  (item) => {
    expect(fromFont[item]).toMatchSnapshot();
  }
);
