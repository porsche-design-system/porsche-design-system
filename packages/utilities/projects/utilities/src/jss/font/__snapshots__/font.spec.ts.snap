// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain correct values for font 1`] = `
Object {
  "family": "'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
  "size": Object {
    "large": Object {
      "fontSize": "2.25rem",
      "lineHeight": 1.3333333333,
    },
    "medium": Object {
      "fontSize": "1.5rem",
      "lineHeight": 1.5,
    },
    "small": Object {
      "fontSize": "1rem",
      "lineHeight": 1.5,
    },
    "xLarge": Object {
      "fontSize": "3.25rem",
      "lineHeight": 1.2307692308,
    },
    "xSmall": Object {
      "fontSize": "0.75rem",
      "lineHeight": 1.6666666667,
    },
  },
  "weight": Object {
    "bold": 700,
    "regular": 400,
    "semibold": 600,
    "thin": 100,
  },
}
`;

exports[`should contain correct values for fontFamily 1`] = `"'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif"`;

exports[`should contain correct values for fontSize 1`] = `
Object {
  "large": Object {
    "fontSize": "2.25rem",
    "lineHeight": 1.3333333333,
  },
  "medium": Object {
    "fontSize": "1.5rem",
    "lineHeight": 1.5,
  },
  "small": Object {
    "fontSize": "1rem",
    "lineHeight": 1.5,
  },
  "xLarge": Object {
    "fontSize": "3.25rem",
    "lineHeight": 1.2307692308,
  },
  "xSmall": Object {
    "fontSize": "0.75rem",
    "lineHeight": 1.6666666667,
  },
}
`;

exports[`should contain correct values for fontWeight 1`] = `
Object {
  "bold": 700,
  "regular": 400,
  "semibold": 600,
  "thin": 100,
}
`;
